Sources/Datadog/Datadog.swift

```diff
@@ -331,6 +331,10 @@ public class Datadog {
     internal static func internalFlushAndDeinitialize() {
         assert(Datadog.isInitialized, "SDK must be first initialized.")
 
+        // Flush the context provider's work items
+        let core = defaultDatadogCore as? DatadogCore
+        core?.contextProvider.queue.sync(flags: .barrier) { }
```

Comment:
The queue is actually serial, so the `.barrier` will have no effect. It's a safeguard if we ever change it to concurrent.